### PR TITLE
RDK-36410:RDKServices doc: Update JSON Definition

### PR DESCRIPTION
**Pull request by pkumbh631**

### CODE REVIEW

File: DeviceDiagnostics/DeviceDiagnostics.cpp [hunk 1]
```diff
@@ -259,8 +263,107 @@ namespace WPEFramework
...
+         *      "success":<bool>}
+         * @return      : Core::<StatusCode>
+         */
+        uint32_t DeviceDiagnostics::getMilestones(const JsonObject& parameters,
+                JsonObject& response)
+        {
+            bool retAPIStatus = false;
+	    std::vector<string> milestones;
+
+            if (Utils::fileExists(MILESTONES_LOG_FILE)) {
```
Comment (npoltorapavlo) [inline DeviceDiagnostics/DeviceDiagnostics.cpp:279]: use `Core::File(MILESTONES_LOG_FILE).Exists()`

File: DeviceDiagnostics/DeviceDiagnostics.h [hunk 2]
```diff
@@ -27,6 +27,13 @@
 
 #include "Module.h"
 
+#define MILESTONES_LOG_FILE                     "/opt/logs/rdk_milestones.log"
+
+enum SysSrv_ErrorCode {
+    SysSrv_FileNotPresent,
+    SysSrv_FileAccessFailed
+};
+
```
Comment (npoltorapavlo) [inline DeviceDiagnostics/DeviceDiagnostics.h:36]: please move to the cpp

File: DeviceDiagnostics/DeviceDiagnostics.h [hunk 3]
```diff
@@ -74,4 +82,36 @@ namespace WPEFramework {
...
+void setJSONResponseArray(JsonObject& response, const char* key,
+        const std::vector<string>& items);
+
+/***
+ * @brief   : Used to construct response with module error status.
+ * @param1[in]  : Error Code
+ * @param2[out]: "response" JSON Object which is returned by the API
+   with updated module error status.
+ */
+void populateResponseWithError(int errorCode, JsonObject& response);
```
Comment (npoltorapavlo) [inline DeviceDiagnostics/DeviceDiagnostics.h:116]: please remove

**Review by npoltorapavlo — CHANGES_REQUESTED** [review]

File: DeviceDiagnostics/DeviceDiagnostics.h (hunk 3 above)
Comment (pkumbh631) [reply DeviceDiagnostics/DeviceDiagnostics.h:109]: @npoltorapavlo can you plz tell me why u want me to remove these declaration?

**Review by anand-ky — APPROVED** [review]

**Pull request merged**